tests/devices/default_qubit/test_default_qubit_native_mcm.py

```diff
@@ -535,3 +562,58 @@ def func(x, y):
     grad2 = qml.grad(func2)(*param)
 
     assert np.allclose(grad1, grad2, atol=0.01, rtol=0.3)
+
+
+@flaky(max_runs=5)
+@pytest.mark.parametrize("shots", [5000, [5000, 5001]])
+@pytest.mark.parametrize("postselect", [None, 0, 1])
+@pytest.mark.parametrize("reset", [False, True])
+@pytest.mark.parametrize("measure_fn", [qml.expval, qml.sample, qml.probs, qml.counts])
+def test_broadcasting_qnode(shots, postselect, reset, measure_fn):
+    """Test that executing qnodes with broadcasting work as expected"""
+    if measure_fn is qml.sample and postselect is not None:
+        pytest.skip("Postselection with samples doesn't work with broadcasting")
+
+    dev = qml.device("default.qubit", shots=shots)
+    param = [[np.pi / 3, np.pi / 4], [np.pi / 6, 2 * np.pi / 3]]
+    obs = qml.PauliZ(0) @ qml.PauliZ(1)
+
+    @qml.qnode(dev)
+    def func(x, y):
+        obs_tape(x, y, None, reset, postselect)
+        return measure_fn(op=obs)
+
+    func1 = func
+    func2 = qml.defer_measurements(func)
+
+    results1 = func1(*param)
+    results2 = func2(*param)
+
+    if measure_fn is qml.sample and postselect is None:
+        if isinstance(shots, list):
+            for s, r1, r2 in zip(shots, results1, results2):
+                assert len(r1) == len(r2) == s
+        else:
+            assert len(results1) == len(results2) == shots
+
+    validate_measurements(measure_fn, shots, results1, results2, batch_size=2)
+
+
+def test_sample_with_broadcasting_and_postselection_error():
+    """Test that an error is raised if returning qml.sample if postselecting with broadcasting"""
+    tape = qml.tape.QuantumScript(
+        [qml.RX([0.1, 0.2], 0), MidMeasureMP(0, postselect=1)], [qml.sample(wires=0)], shots=10
+    )
+    dev = qml.device("default.qubit", shots=10)
+
+    with pytest.raises(ValueError, match="Returning qml.sample is not supported when"):
+        qml.transforms.dynamic_one_shot(tape)
```

Comment:
For locality, maybe :shrug:
```suggestion
    with pytest.raises(ValueError, match="Returning qml.sample is not supported when"):
        qml.transforms.dynamic_one_shot(tape)
        
    dev = qml.device("default.qubit", shots=10)
```